mokapot/confidence.py

```diff
@@ -327,53 +235,70 @@ class LinearConfidence(Confidence):
 
     Parameters
     ----------
-    psms : LinearPsmDataset object
+    psms : OnDiskPsmDataset
```

Comment:
From how I understand the code right now, `LinearPsmDataset` is the in-memory representation of a chunk of the input data (and associated utilities functions for example to easily get all targets and decoys), whereas `OnDiskPsmDataset` is a file pointer to the data on disk.

The `brew._create_psms` function instantiates a `LinearPsmDataset` from a given `OnDiskPsmDataset` object and one chunk of the same `OnDiskPsmDataset`. 

The function you mentioned works directly with `OnDiskPsmDataset`.

Reply:
I think your question is spot on, and very much highlights that we should change naming of the classes, so that they reflect their function and how they are used.